rwengine/include/engine/GameInputState.hpp

```diff
@@ -0,0 +1,58 @@
+#ifndef RWENGINE_GAMEINPUTSTATE_HPP
+#define RWENGINE_GAMEINPUTSTATE_HPP
+
+struct GameInputState
+{
+	/// Inputs that can be controlled
+	/// @todo find any sensible values
+	enum Control {
+		/* On Foot */
+		FireWeapon = 0,
+		NextWeapon,  // Also Next Target
+		LastWeapon,  // Also Last Target
+		GoForward,
+		GoBackwards,
+		GoLeft,
+		GoRight,
+		ZoomIn,
+		ZoomOut,
+		EnterExitVehicle,
+		ChangeCamera,
+		Jump,
+		Sprint,
+		AimWeapon,
+		Crouch,
+		Walk,
+		LookBehind,
+
+		/* In Vehicle */
+		VehicleFireWeapon,
```

Comment:
These indices should be the same as the on-foot slots?
If you don't want to do it this way you'll have to seperate onfoot and vehicle controls more cleanly (adding a seperator enum and guaranteeing the order).

The script checks the buttons by index. So I'd assume these indices are equivalent to these at all times (both, onfoot and in vehicle).